templates/centos-8/nfs.conf.erb

```diff
@@ -0,0 +1,19 @@
+# Generated by Chef for <%= node['fqdn'] %>
```

Comment:
how did we come up with this config? can you add some details?

Reply:
I can see there are some differences with the default config for example in teh fact that lockd only has the udp port configured

Reply:
I was trying to translate all options set in templates used by NFS cookbook into options in `nfs.conf`

Reply:
you mean from 
```
<% when 'rhel' -%>
# Rendered RHEL template variant
STATD_PORT="<%= node['nfs']['port']['statd'] -%>"
STATD_OUTGOING_PORT="<%= node['nfs']['port']['statd_out'] -%>"
STATDARG="-p <%= node['nfs']['port']['statd'] -%> -o <%= node['nfs']['port']['statd_out'] -%>"
MOUNTD_PORT="<%= node['nfs']['port']['mountd'] -%>"
RPCMOUNTDOPTS="-p <%= node['nfs']['port']['mountd'] -%>"
LOCKD_UDPPORT="<%= node['nfs']['port']['lockd'] -%>"
LOCKD_TCPPORT="<%= node['nfs']['port']['lockd'] -%>"
RQUOTAD_PORT="<%= node['nfs']['port']['rquotad'] -%>"
  <% unless node['nfs']['v2'].nil? -%>
MOUNTD_NFS_V2="<%= node['nfs']['v2'] -%>"
  <% end -%>
  <% unless node['nfs']['v3'].nil? -%>
MOUNTD_NFS_V3="<%= node['nfs']['v3'] -%>"
  <% end -%>
RQUOTAD="<%= node['nfs']['rquotad'] -%>"
  <% unless node['nfs']['threads'] == 0 -%>
RPCNFSDCOUNT="<%= node['nfs']['threads'] -%>"
  <% end -%>
```

Can you please add some details on why you excluded some of them (such as `LOCKD_TCPPORT`  `RQUOTAD_PORT` `RQUOTAD`)?

Reply:
Setting `[lockd] - port` should be equivalent to setting both `LOCKD_TCPPORT` and `LOCKD_UDPPORT` from old config
From [doc of `nfs.conf`](https://man7.org/linux/man-pages/man5/nfs.conf.5.html) and [doc of `rpc.statd`](https://man7.org/linux/man-pages/man8/rpc.statd.8.html):
> lockd  Recognized values: port and udp-port.

>-T, --nlm-port port
              Specifies the port number that lockd should listen on for NLM
              requests.  This sets both the TCP and UDP ports unless the UDP
              port is set separately.
...
The values recognized in the [lockd] section include port and udp-port which have the same effect as the --nlm-port and --nlm-udp-port options, respectively.

There is no option to configure `rquotad` in `nfs.conf` anymore.
From [this guide](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/deploying_different_types_of_servers/exporting-nfs-shares_deploying-different-types-of-servers) it seems there is still a way to define the `rquotad` port number:
> You can override the default port number by appending -p port-number to the RPCRQUOTADOPTS variable in the /etc/sysconfig/rpc-rquotad file.

But I would argue this is not needed, since nfs recipe seems to [disable `rquotad` by default](https://github.com/atomic-penguin/cookbook-nfs/blob/master/attributes/default.rb#L27)